fsspec/implementations/ftp.py

```diff
@@ -166,6 +168,13 @@ def mv(self, path1, path2, **kwargs):
     def __del__(self):
         self.ftp.close()
 
+    def invalidate_cache(self, path=None):
+        if path and path in self.dircache:
+            del self.dircache[path]
+        else:
+            self.dircache.clear()
```

Comment:
It seems that this gets called if the path passed happens not to be int he cache. It should only be called if path is None
```
if path is None:
    self.dircache.clear()
else:
    self.dircache.pop(path, None)'
```

Reply:
Yes, good catch! I have fixed this bug and another one in the `_parent` method by the way. Please take a look again.